### PR TITLE
Sonar: 'dayjs/esm/locale/en' import is duplicated

### PR DESCRIPTION
**Pull request by qmonmert**
Fix https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&id=jhipster-sample-application&open=AYYImbn4jT0F0g-yuSk2

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/client/support/update-languages.mts [hunk 1]
```diff
@@ -37,7 +37,8 @@ export function updateLanguagesInDayjsConfigurationTask(
   const { languagesDefinition = [] } = application;
   const { ignoreNeedlesError: ignoreNonExisting } = control;
 
-  const newContent = languagesDefinition.reduce(
+  const languagesDefinitionNoDuplicates = [...new Map(languagesDefinition.map(v => [v.dayjsLocale, v])).values()];
```
Comment (mshima) [inline generators/client/support/update-languages.mts:40]:
```suggestion
  const languagesDefinitionNoDuplicates = [...new Set(languagesDefinition.map(v => v.dayjsLocale))];
```

**Review by mshima — COMMENTED** [review]
Using a set is easier to understand.

File: generators/client/support/update-languages.mts (hunk 1 above)
Comment (mshima) [reply generators/client/support/update-languages.mts:40]: Won’t work

Comment (mshima) [reply generators/client/support/update-languages.mts:40]: Need to below change

File: generators/client/support/update-languages.mts [hunk 2]
```diff
@@ -37,7 +37,8 @@ export function updateLanguagesInDayjsConfigurationTask(
...
+  const newContent = languagesDefinitionNoDuplicates.reduce(
     (content, language) => `${content}import 'dayjs/${commonjs ? '' : 'esm/'}locale/${language.dayjsLocale}'\n`,
```
Comment (mshima) [inline generators/client/support/update-languages.mts:42]:
```suggestion
    (content, dayjsLocale) => `${content}import 'dayjs/${commonjs ? '' : 'esm/'}locale/${dayjsLocale}'\n`,
```

File: generators/client/support/update-languages.mts (hunk 1 above)
Comment (mshima) [inline generators/client/support/update-languages.mts:40]:
```suggestion
  const uniqueDayjsLocales = [...new Map(languagesDefinition.map(v => [v.dayjsLocale, v])).values()];
```

File: generators/client/support/update-languages.mts [hunk 3]
```diff
@@ -37,7 +37,8 @@ export function updateLanguagesInDayjsConfigurationTask(
   const { languagesDefinition = [] } = application;
   const { ignoreNeedlesError: ignoreNonExisting } = control;
 
-  const newContent = languagesDefinition.reduce(
+  const languagesDefinitionNoDuplicates = [...new Map(languagesDefinition.map(v => [v.dayjsLocale, v])).values()];
+  const newContent = languagesDefinitionNoDuplicates.reduce(
```
Comment (mshima) [inline generators/client/support/update-languages.mts:41]:
```suggestion
  const newContent = uniqueDayjsLocales.reduce(
```

File: generators/client/support/update-languages.mts (hunk 2 above)
Comment (qmonmert) [reply generators/client/support/update-languages.mts:42]: it doesn't work

File: generators/client/support/update-languages.mts (hunk 1 above)
Comment (qmonmert) [reply generators/client/support/update-languages.mts]: done

**Review by mshima — APPROVED** [review]

**Pull request merged**